4)	ABC is an online advertisement firm, the management found adblocker used by the users is affecting the revenue. The firm has invested significantly to distribute custom adblock whitelists. Below is the table showing the number of sample adblock detected before the distribution of the first 3 months (12 weeks) and the next 3 months after the whitelist has been distributed. (Ideally use jupyter notebook)

# Table 1: Before Distribution
|Week | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12 |
|-----|----|----|----|----|----|----|----|----|----|----|----|----|
|Count| 3100 | 2800 | 1900 | 2400 | 3200 | 2700 | 1600 | 4100 | 2300 | 3200 | 2900 | 3300 |


# Table 2: After Distribution
|Week | 13 | 14 | 15 | 16 | 17 | 18 | 19 | 20 | 21 | 22 | 23 | 24 |
|-----|----|----|----|----|----|----|----|----|----|----|----|----|
|Count| 2300 | 2100 | 1900 | 2400 | 3500 | 1700 | 1800 | 2400 | 3300 | 2700 | 2100 | 2300 |

a)	Convert the above information into a numpy array and compute the average difference between the first table and the second table.

b)	Could the difference legit and not due to chance? Use a statistical approach to demonstrate your assumptions.

c)	Because the firm already spent vast amounts of money, is the difference in the number of adblocker detected for the next 3 months statistically significant and worth the investment? You may provide your opinion here.

d)	What’s the confidence intervals of the average difference? You may use 90% or 95% percentile for confidence intervals, state this percentile in the solution.


---
# **ANSWER**

---
a)	Convert the above information into a numpy array and compute the average difference between the first table and the second table.



In [1]:
import numpy as np
import pandas as pd

before = np.array([3100,2800,1900,2400,3200,2700,1600,4100,2300,3200,2900,3300]) # table 1
after = np.array([2300,2100,1900,2400,3500,1700,1800,2400,3300,2700,2100,2300]) # table 2

In [2]:
# computing the average of 2 tables

avg_before = np.average(before)
avg_after = np.average(after)

# print the difference as 2 decimal places
print("Average Difference Between The First Table and The Second Table: " + str(round(avg_before - avg_after,2)))

Average Difference Between The First Table and The Second Table: 416.67


Average Difference Between The First Table and The Second Table: **416.67**

---
b) Could the difference legit and not due to chance? Use a statistical approach to demonstrate your assumptions.


To establish statistical significance we have to come up with a ***null hypothesis*** and an ***alternative hypothesis***. A ***null hypothesis*** is where the variable has no impact on our end result. The ***alternative hypothesis*** is where the variable does have an impact on the end result.

For this problem, the null and alternative hypothesis are:

* Null Hypothesis, $H_{0}$ : The average difference of sample adblock detected after the custom adblock whitelist distribution is the same as before distribution (i.e.: the average difference, mean = 0).

* Alternative Hypothesis, $H_{1}$ : The average difference of sample adblock detected after the custom adblock whitelist distribution increases distribution (mean > 0).

Approach:
* Significance level of 0.05 (5%) is chosen as the common standard for hypothesis testings.
* T-test (one-tailed)

T-test (one-tailed) is chosen because:
* The question is dealing with determining whether a solution has effect on the population of interest.
* We are comparing the means of 2 groups.
* The sample size, n is small (n < 30)
* We are only concerned with the increase of average difference.

In [3]:
from scipy.stats import ttest_1samp
stats,p_value = ttest_1samp(before - after, popmean = 0)
print("2-tailed p-value= " + str(round(p_value,2)))
print("1-tailed p-value= " + str(round(p_value/2,2)))

2-tailed p-value= 0.08
1-tailed p-value= 0.04


Intepretation:

$p-value\ \  (0.04) < 0.05$ are satistically significant ($H_{0}$ is rejected, $H_{1}$ is accepted).

Therefore, the difference is ***not due to chance*** (randomness) with confidence interval of 95% (1-significance level).

---
c) Because the firm already spent vast amounts of money, is the difference in the number of adblocker detected for the next 3 months statistically significant and worth the investment? You may provide your opinion here.


As answered in the previous question, the average difference in the number of adblocker detected is ***statistically significant***. However, the approach of analysing collected data (provided) week by week for this hypothesis testing ***overlooked the effect of seasonality***. I would suggest comparing the average difference before and after distribution during same or similar season.

---
d) What’s the confidence intervals of the average difference? You may use 90% or 95% percentile for confidence intervals, state this percentile in the solution.


The confidence interval of the average difference for this question is $95\%$.

---